presto-main/src/main/java/com/facebook/presto/operator/aggregation/ArrayAggregation.java

```diff
@@ -0,0 +1,202 @@
+    {
+        Type valueType = types.get("T");
+        Signature signature = new Signature(NAME,
+                parameterizedTypeName("array", valueType.getTypeSignature()),
+                valueType.getTypeSignature());
+        InternalAggregationFunction aggregation = generateAggregation(valueType);
+        return new FunctionInfo(signature, getDescription(), aggregation.getIntermediateType().getTypeSignature(), aggregation, false);
+    }
+
+    private static InternalAggregationFunction generateAggregation(Type valueType)
+    {
+        DynamicClassLoader classLoader = new DynamicClassLoader(ArrayAggregation.class.getClassLoader());
+
+        ArrayAggregationStateSerializer stateSerializer = new ArrayAggregationStateSerializer();
+        Type intermediateType = stateSerializer.getSerializedType();
+
+        List<Type> inputTypes = ImmutableList.of(valueType);
+        Type outputType = new ArrayType(valueType);
+        ArrayAggregationStateFactory stateFactory = new ArrayAggregationStateFactory(valueType);
+        AggregationMetadata metadata = new AggregationMetadata(
+                generateAggregationName(NAME, valueType, inputTypes),
+                createInputParameterMetadata(valueType),
+                INPUT_FUNCTION,
+                null,
+                null,
+                COMBINE_FUNCTION,
+                OUTPUT_FUNCTION,
+                ArrayAggregationState.class,
+                stateSerializer,
+                stateFactory,
+                outputType,
+                false);
+
+        GenericAccumulatorFactoryBinder factory = new AccumulatorCompiler().generateAccumulatorFactoryBinder(metadata, classLoader);
+        return new InternalAggregationFunction(NAME, inputTypes, intermediateType, valueType, true, false, factory);
+    }
+
+    private static List<ParameterMetadata> createInputParameterMetadata(Type value)
+    {
+        return ImmutableList.of(new ParameterMetadata(STATE), new ParameterMetadata(INPUT_CHANNEL, value), new ParameterMetadata(BLOCK_INDEX));
+    }
+
+    public static void input(ArrayAggregationState state, Block value, int position)
+    {
+        if (state.getSliceOutput() == null) {
+            SliceOutput sliceOutput = new DynamicSliceOutput(400);
+            state.setEntries(0);
+            state.setSliceOutput(sliceOutput);
+        }
+        state.setEntries(state.getEntries() + 1);
+        appendTo(state.getType(), state.getSliceOutput(), value, position);
+    }
+
+    private static void appendTo(Type type, SliceOutput output, Block block, int position)
+    {
+        if (type.getJavaType() == long.class) {
+            output.appendLong(type.getLong(block, position));
+        }
+        else if (type.getJavaType() == double.class) {
+            output.appendDouble(type.getDouble(block, position));
+        }
+        else if (type.getJavaType() == Slice.class) {
+            Slice s = type.getSlice(block, position);
```

Comment:
We generally try not to abbreviate variable names because it tends to make code harder to read